### ISSUE TITLE
Ensime should not recommend Coursier as alternative to Ivy with a warning

### ISSUE THREAD

**Issue by alexandru** [ensime/ensime-sbt#206]
While using Ensime's SBT plugin, I noticed this warning:

```
[warn] SBT is using ivy to resolve dependencies and is known to be slow. Coursier is recommended: http://get-coursier.io
```

I personally take warnings very seriously, which means that when I see a marketing message like this one, I expect for that alternative to be a good and stable alternative. So I tried Coursier. And ALAS it crashed with an exception while updating the Ensime server from within Emacs and it was Coursier's fault, as it was a Coursier related exception and the server was updated just fine with Ivy.

So I'm all for the cutting edge, but marketing messages for experimental packages shouldn't be placed in SBT warnings imho. If you want to make this recommendation, the far better place is in Ensime's documentation.

**Comment by fommil** [comment]
well, I disagree

**Comment by fommil** [comment]
We used to get a flood of complaints about sbt-ensime being really slow, so this is to stop people from raising those tickets. Change it to `info` if you must.

**Comment by fommil** [comment]
coursier isn't used to update the server inside emacs.

**Comment by alexandru** [comment]
@fommil I understand the rationale, however it's far worse for Ensime to crash than it is for it to be slow, imho.

Please see the updated description.

**Comment by fommil** [comment]
just add coursier on a per project basis, not system wide. If Alex is recommending to install globally, please send a PR to update his docs. If you would like to use coursier to upgrade ensime-server itself, please help out by picking up https://github.com/ensime/ensime-emacs/issues/269

I'm not disabling this log message, although I would accept a PR to turn it into an `info` if you feel strongly about it.

**Comment by fommil** [comment]
@alexarchambault I'm not sure coursier is ready for global installation as per https://github.com/alexarchambault/coursier#sbt-plugin

I recommend that you only document how to install on a per-project basis.

**Comment by fommil** [comment]
also looks like @alexandru has a dodgy maven cache.

**Comment by alexandru** [comment]
I personally *never* add IDE-related or other tooling plugins on a project level, because I do not want to force that choice upon the people that may want to build the project. Adding Coursier to `project/plugins.sbt` would be just like adding `ensime` to it.

Also, my Maven cache works fine without Coursier.

**Comment by fommil** [comment]
well, in either case this sounds like a bug report for coursier. I don't consider coursier to be IDE tooling, it's third party dependency management and is *very much* a project-level concern.

**Comment by alexarchambault** [comment]
@alexandru It seems you somehow restricted the repositories to your `~/.m2`. Unlike SBT/Ivy, coursier doesn't mix the caches (under `~/.coursier/cache`) of the various repositories, so if a given repository isn't setup in a resolution, its cached content just won't be used. Hence the possible discrepancy you're seeing.

**Comment by alexandru** [comment]
@alexarchambault thanks, maybe that's the problem, I'll retry.

**Comment by fommil** [comment]
if you need to reproduce this, you could start with the `build.sbt` that we write out and execute in ensime-emacs https://github.com/ensime/ensime-emacs/blob/master/ensime-startup.el#L28-L72

**Comment by fommil** [comment]
FYI we do add the local maven repo, so that we can see local builds of scala.

I just noticed that we still have `fullResolvers += Resolver.jcenterRepo`... which we don't need anymore. @alexandru a PR to remove that line as your penance would be appreciated :stuck_out_tongue_closed_eyes:

**Comment by howard-wang-nl** [comment]
I tried the sbt coursier plugin. It causes a problem with test jar dependency resolving.  If the test task of project A depends on test jar of project B, `sbt test` of project A will fail to resolve symbols in test jar of project B.  Ivy2 doesn't have this problem.  This bug was reported https://github.com/alexarchambault/coursier/issues/318 .

So I agree with @alexandru on this discussion.

**Comment by fommil** [comment]
@howardlunatech you should file a reproduction with coursier in that case. Much to Alex's credit, he has fixed the bug, so I don't see what the problem is.

**Comment by howard-wang-nl** [comment]
@fommil Seems already reported and fixed in https://github.com/alexarchambault/coursier/pull/373 but not released to public repo yet?  At least I followed the doc of ensime-vim to setup and this problem wasted me a lot time to track down the cause.  So it will in effect drive users off Ensime, thus not good to advocate Coursier in this stage yet.

**Comment by fommil** [comment]
you seem to mistake our intentions, our goal is not to get more users of ensime... it is to make a free/libre scala editing platform that give our users the freedom to hack their editor. From the main website:

> ENSIME is free/libre and appeals to hackers, minimalists and connoisseurs - engineers and artists who craft their own exquisite tools and help their neighbour. If you would rather use a more mainstream IDE, you might prefer IntelliJ IDEA or Scala IDE. We don’t claim to be better, just different.

**Comment by alexarchambault** [comment]
@howardlunatech FYI https://github.com/alexarchambault/coursier/pull/373 is included in coursier `1.0.0-M14-7` and `1.0.0-M14-8`.
